### DISCUSSION TITLE
Notification as long as Omnisearch is indexing

### DISCUSSION THREAD

**Discussion by zsolt-juhasz** [scambier/obsidian-omnisearch#354] [Ideas]
I have a very large vault - 8.000+ notes and 5.000+ PDFs - which all get indexed. You can imagine, this will take a while.

Each time I start Obsidian, I get the notification that Omnisearch is indexing the vault and then dissapeares.

![image](https://github.com/scambier/obsidian-omnisearch/assets/31696691/354fdec3-b6b0-4d82-ae85-a9773e54c5c5)

During this time, I can't really use Obsidian, as it is lagging, not opening files and so on. 

It would be great to have a additional permanent notification - maybe in the statusbar - which shows that OS is still indexing (even better with a progress bar or time-left indicator. 

As OS already shows a notification on start and on end of indexing - it should be "easy" to implement such information on the statusbar.

What is your opinion about it?

**Comment by scambier** [comment]
First, you should make sure that Omnisearch caches its index.
![image](https://github.com/scambier/obsidian-omnisearch/assets/3216752/f3f5e427-90bf-4ac9-a263-1e565037c73a)
(This setting will disable itself if you close Obsidian during the cache write operation, to avoid bootloop crashes.)

Then you have to let it run to completion. You can check if Omnisearch is working when opening the modal; there will be a "⏳ Work in progress" message under the search field. Once the indexing is done and saved, it should just freeze for a few seconds at startup while reloading the cache.

Now to answer your question, it should be doable to add a progress bar in addition to the message, in the Omnisearch modal.
